### PR TITLE
BDM Documentation

### PR DESCRIPTION
**Pull request by alistairbntl**
Swapped the negative sign in 2D BDM basis functions to be consistent with the definition of scalar function curl.  The change in sign does not change the polynomial space.

Added some documentation to BDM polynomial description to define scalar function curl.

Corrected the pressure space documenation for BDM elements.

### CODE REVIEW

File: source/base/polynomials_bdm.cc [hunk 1]
```diff
@@ -117,8 +117,8 @@ PolynomialsBDM<dim>::compute (const Point<dim>            &unit_point,
         {
           values[start][0] = monovali[0][0];
           values[start][1] = -unit_point(1) * monovali[0][1];
-          values[start+1][0] = -unit_point(0) * monovali[1][1];
-          values[start+1][1] = monovali[1][0];
+          values[start+1][0] = unit_point(0) * monovali[1][1];
+          values[start+1][1] = -monovali[1][0];
```
Comment (bangerth) [inline source/base/polynomials_bdm.cc]: If you change the values here, wouldn't you also have to change the gradients and second derivatives below?

Comment (alistairbntl) [inline source/base/polynomials_bdm.cc]: Of course, thanks for noticing.

**Comment by bangerth** [comment]
@guidokanschat : I think you know this element best. Can you please take care of this PR?

**Comment by tjhei** [comment]
The tests fe/bdm* are now working again with 35938c6

**Comment by bangerth** [comment]
@guidokanschat ?

**Pull request merged**